src/components/contextual/pages/pool/invest/InvestPage.vue

```diff
@@ -0,0 +1,195 @@
+<script setup lang="ts">
+import { onBeforeMount, computed } from 'vue';
+import useInvestState from '@/components/forms/pool_actions/InvestForm/composables/useInvestState';
+import InvestForm from '@/components/forms/pool_actions/InvestForm/InvestForm.vue';
+import TradeSettingsPopover, {
+  TradeSettingsContext,
+} from '@/components/popovers/TradeSettingsPopover.vue';
+import usePoolTransfers from '@/composables/contextual/pool-transfers/usePoolTransfers';
+import { usePool } from '@/composables/usePool';
+import { forChange, isSameAddress, indexOfAddress } from '@/lib/utils';
+import { configService } from '@/services/config/config.service';
+import InvestFormV2 from '@/components/forms/pool_actions/InvestForm/InvestFormV2.vue';
+import BalAccordion from '@/components/_global/BalAccordion/BalAccordion.vue';
+import MyWallet from '@/components/cards/MyWallet/MyWallet.vue';
+import useBreakpoints from '@/composables/useBreakpoints';
+
+import StakingProvider from '@/providers/local/staking/staking.provider';
+
+import useNativeBalance from '@/composables/useNativeBalance';
+import useTokens from '@/composables/useTokens';
+import useJoinPool from '@/composables/pools/useJoinPool';
+import useInvestPageTabs, { tabs } from '@/composables/pools/useInvestPageTabs';
+
+/**
+ * COMPOSABLES
+ */
+const { network } = configService;
+const { pool, loadingPool, transfersAllowed, useNativeAsset } =
+  usePoolTransfers();
+const { isDeepPool, isWethPool } = usePool(pool);
+const { sor, sorReady } = useInvestState();
+const { activeTab } = useInvestPageTabs();
+const { tokenAddresses, amounts } = useInvestState();
+const { upToLargeBreakpoint } = useBreakpoints();
+const { addTokensIn, setAmountsIn, isSingleAssetJoin, amountsIn } =
+  useJoinPool();
```

Comment:
Does this not break when the pool isn't a deep pool because you don't use the provider: https://github.com/balancer-labs/frontend-v2/pull/2339/files#diff-de8992b55fdef033962c2ab6870a3b4ae1e1f9bba6309a660120f32bc5e82049R25

Reply:
It works, but all these properties of `useJoinPool` will be `undefined`.

I moved the logic to it's own component:
`src/components/contextual/pages/pool/invest/InvestPageMyWallet.vue`